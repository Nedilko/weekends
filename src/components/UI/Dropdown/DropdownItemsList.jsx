import GeneralPanel from '@components/UI/GeneralPanel'
import PropTypes from 'prop-types'

DropdownItemsList.propTypes = {
  selectedValue: PropTypes.string,
  className: PropTypes.string,
  items: PropTypes.array,
  handleSelect: PropTypes.func,
}

function DropdownItemsList({ items, selectedValue, handleSelect, className }) {
  return (
    <GeneralPanel className={className}>
      {items.map((item, index) => (
        <option
          onClick={(e) => handleSelect(e.target.value)}
          key={index}
          value={item}
          className={`px-2 py-1 text-sm hover:bg-slate-200 dark:hover:bg-slate-700 ${
            selectedValue === item ? 'font-regular' : 'font-thin'
          }`}
        >
          {item}
        </option>
      ))}
    </GeneralPanel>
  )
}

export default DropdownItemsList
