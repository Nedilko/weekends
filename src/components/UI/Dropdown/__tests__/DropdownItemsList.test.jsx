import DropdownItemsList from '@UI/Dropdown/DropdownItemsList'
import { render, screen, userEvent } from '@utils/test-utils'

jest.mock('@UI/Dropdown/DropdownItem', () => {
  const originalModule = jest.requireActual('@UI/Dropdown/DropdownItem')
  return {
    __esModule: true,
    ...originalModule,
    default: ({ value, handleSelect }) => (
      <div data-testid="dropdown-item" onClick={() => handleSelect(value)}>
        {value}
      </div>
    ),
  }
})

describe('DropdownItemsList', () => {
  const selectHandler = jest.fn()

  beforeEach(() => {
    selectHandler.mockClear()
  })

  it('should render dropdown items list', () => {
    const selectedValue = 'second'
    const { container } = render(
      <DropdownItemsList
        items={['first']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render 3 dropdown items', () => {
    const selectedValue = 'second'
    render(
      <DropdownItemsList
        items={['first', 'second', 'third']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    expect(screen.getAllByTestId('dropdown-item')).toHaveLength(3)
  })

  it('should handle select', async () => {
    const selectedValue = 'second'
    const user = userEvent.setup()
    render(
      <DropdownItemsList
        items={['first', 'second', 'third']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    await user.click(screen.getByText('third'))
    expect(selectHandler).toHaveBeenCalledTimes(1)
  })

  it('should handle select with dropdown item value', async () => {
    const selectedValue = 'second'
    const user = userEvent.setup()
    render(
      <DropdownItemsList
        items={['first', 'second', 'third']}
        handleSelect={selectHandler}
        selectedValue={selectedValue}
      />
    )
    await user.click(screen.getByText('third'))
    expect(selectHandler).toHaveBeenCalledWith('third')
  })
})
