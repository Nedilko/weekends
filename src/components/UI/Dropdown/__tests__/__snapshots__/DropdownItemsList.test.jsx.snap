// Vitest Snapshot v1

exports[`DropdownItemsList > should render dropdown items list 1`] = `
<div>
  <div
    class="absolute overflow-auto rounded-md bg-white shadow-md dark:bg-zinc-900 "
  >
    <div
      data-testid="dropdown-item"
    >
      first
    </div>
  </div>
</div>
`;

exports[`DropdownItemsList should render dropdown items list 1`] = `
<div>
  <div
    class="absolute overflow-auto rounded-md bg-white shadow-md dark:bg-zinc-900 "
  >
    <div
      data-testid="dropdown-item"
    >
      first
    </div>
  </div>
</div>
`;
